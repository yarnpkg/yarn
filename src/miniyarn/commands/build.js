import {flags} from '@manaflair/concierge';

export default concierge =>
  concierge
    .command(`build [... args]`)
    .describe(`Run the project build script, if available`)
    .flag(flags.PROXY_COMMAND)
    .action(async args => {
      return await concierge.run(null, [`run`, `build`, ...args.args], args);
    });
