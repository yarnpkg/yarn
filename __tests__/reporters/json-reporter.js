/* @flow */
/* eslint quotes: 0 */

import type {MockData} from "./_mock.js";
import JSONReporter from "../../src/reporters/json-reporter.js";
import build from "./_mock.js";

const getJSONBuff = build(JSONReporter, (data): MockData => data);

test("JSONReporter.step", async () => {
  expect(await getJSONBuff((r) => {
    r.step(1, 5, "foobar");
  })).toMatchSnapshot();
});

test("JSONReporter.footer", async () => {
  expect(await getJSONBuff((r) => {
    r.footer(false);
  })).toMatchSnapshot();
});

test("JSONReporter.log", async () => {
  expect(await getJSONBuff((r) => {
    r.log("foobar");
  })).toMatchSnapshot();
});

test("JSONReporter.command", async () => {
  expect(await getJSONBuff((r) => {
    r.command("foobar");
  })).toMatchSnapshot();
});

test("JSONReporter.success", async () => {
  expect(await getJSONBuff((r) => {
    r.success("foobar");
  })).toMatchSnapshot();
});

test("JSONReporter.error", async () => {
  expect(await getJSONBuff((r) => {
    r.error("foobar");
  })).toMatchSnapshot();
});

test("JSONReporter.warn", async () => {
  expect(await getJSONBuff((r) => {
    r.warn("foobar");
  })).toMatchSnapshot();
});

test("JSONReporter.info", async () => {
  expect(await getJSONBuff((r) => {
    r.info("foobar");
  })).toMatchSnapshot();
});

test("JSONReporter.activity", async () => {
  expect(await getJSONBuff(async function (r): Promise<void> {
    const activity = await r.activity();
    activity.tick("foo");
    activity.tick("bar");
    activity.end();
  })).toMatchSnapshot();
});

test("JSONReporter.progress", async () => {
  expect(await getJSONBuff(async function (r): Promise<void> {
    const tick = await r.progress(2);
    tick();
    tick();
  })).toMatchSnapshot();
});
