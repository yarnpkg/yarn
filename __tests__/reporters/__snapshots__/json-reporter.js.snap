exports[`test JSONReporter.activity 1`] = `
Object {
  "stderr": "",
  "stdout": "{\"type\":\"activityStart\",\"data\":{\"id\":0}}
{\"type\":\"activityTick\",\"data\":{\"id\":0,\"name\":\"foo\"}}
{\"type\":\"activityTick\",\"data\":{\"id\":0,\"name\":\"bar\"}}
{\"type\":\"activityEnd\",\"data\":{\"id\":0}}",
}
`;

exports[`test JSONReporter.command 1`] = `
Object {
  "stderr": "",
  "stdout": "{\"type\":\"command\",\"data\":\"foobar\"}",
}
`;

exports[`test JSONReporter.error 1`] = `
Object {
  "stderr": "{\"type\":\"error\",\"data\":\"foobar\"}",
  "stdout": "",
}
`;

exports[`test JSONReporter.footer 1`] = `
Object {
  "stderr": "",
  "stdout": "{\"type\":\"finished\",\"data\":0}",
}
`;

exports[`test JSONReporter.info 1`] = `
Object {
  "stderr": "",
  "stdout": "{\"type\":\"info\",\"data\":\"foobar\"}",
}
`;

exports[`test JSONReporter.log 1`] = `
Object {
  "stderr": "",
  "stdout": "{\"type\":\"log\",\"data\":\"foobar\"}",
}
`;

exports[`test JSONReporter.progress 1`] = `
Object {
  "stderr": "",
  "stdout": "{\"type\":\"progressStart\",\"data\":{\"id\":0,\"total\":2}}
{\"type\":\"progressTick\",\"data\":{\"id\":0,\"current\":1}}
{\"type\":\"progressTick\",\"data\":{\"id\":0,\"current\":2}}
{\"type\":\"progressFinish\",\"data\":{\"id\":0}}",
}
`;

exports[`test JSONReporter.step 1`] = `
Object {
  "stderr": "",
  "stdout": "{\"type\":\"step\",\"data\":{\"message\":\"foobar\",\"current\":1,\"total\":5}}",
}
`;

exports[`test JSONReporter.success 1`] = `
Object {
  "stderr": "",
  "stdout": "{\"type\":\"success\",\"data\":\"foobar\"}",
}
`;

exports[`test JSONReporter.warn 1`] = `
Object {
  "stderr": "{\"type\":\"warning\",\"data\":\"foobar\"}",
  "stdout": "",
}
`;
